### PR TITLE
feat: [CDS-76800]: Add command line options for terragrunt

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
#### Screenshots

https://github.com/harness/harness-core-ui/assets/98508399/946998d0-35b5-48c6-a1f5-20e0018cd9e7

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/EditView/TerragruntEditView.tsx [hunk 1]
```diff
@@ -806,6 +824,24 @@ export default function TerragruntEditView(
...
+                      <CommandFlags
+                        formik={formik}
+                        stepType={initialValues.type === StepType.TerragruntDestroy ? 'DESTROY' : 'APPLY'}
+                        allowableTypes={allowableTypes}
+                        path={'spec.configuration.commandFlags'}
+                        isTerragrunt={true}
+                      />
+                    }
+                  />
+                </Accordion>
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/EditView/TerragruntEditView.tsx:842]: A variable can be defined to hold this element to reuse here and at line 768.

File: src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx [hunk 2]
```diff
@@ -198,37 +192,54 @@ function TgPlanInputStep(props: TerragruntPlanProps & { formik?: FormikContextTy
...
+          }}
+        />
       )}
+
+      {inputSet?.commandFlags?.map((terragruntCommandFlag: CommandFlags, terragruntFlagIdx: number) => {
+        if (isValueRuntimeInput(get(inputSet, `commandFlags[${terragruntFlagIdx}].flag`))) {
+          return (
+            <>
+              <Label className={css.label}>{getString('cd.commandLineOptions')}</Label>
+              <div key={terragruntFlagIdx}>
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx:227]: `key` should be used at the fragment at line 225. It has to be changed to `<Fragment />` to pass the `key`.

File: src/modules/75-cd/components/PipelineSteps/Common/CommandFlags/CommandFlags.tsx [hunk 3]
```diff
@@ -15,33 +15,67 @@ import { useStrings } from 'framework/strings'
...
+      }
+    : {
+        refetch: refetchTerraform,
+        data: terraformData,
+        loading: terraformLoading
+      }
+
+  React.useEffect(() => {
+    refetch()
+  }, [isTerragrunt])
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/Common/CommandFlags/CommandFlags.tsx:77]: I think this effect can be removed and `isTerragrunt` and `!isTerragrunt` can be used as the `lazy` props of both APIs.

**Review by Eric761 — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Comment by sonaligoyal-harness** [comment]
Coverage is low for existing files.

**Pull request merged**